base/abstractdict.jl

```diff
@@ -527,6 +527,71 @@ function hash(a::AbstractDict, h::UInt)
     hash(hv, h)
 end
 
+"""
+    modify!(f, d::AbstractDict{K, V}, key) -> (old => new)
+
+Lookup and then update, insert or delete in one go without re-computing the hash.
+Return a pair of `old::Union{Some,Nothing}` and `new::::Union{Some,Nothing}`.
+
+`f` is a callable object that must take a single `old::Union{Some, Nothing}` argument and
+return a `new::Union{T, Some{T}, Nothing}` value, where `T` is a type [`convert`](@ref)-able
+to the value type `V`.  The value `Some(d[key])` is passed to `f` if `haskey(d, key)`;
+otherwise `nothing` is passed.  If `f` returns `nothing`, corresponding entry in the
+dictionary `d`  is removed.  If `f` returns non-`nothing` value `x`, `key => something(x)`
+is inserted or updated in `d` (equivalent to `d[key] = something(x)` but more efficient).
+
+Whether `Some{V}(value)` or `Some{typeof(value)}(value)` is returned is an implementation
```

Comment:
It's not clear what value is referred to here, as no single value is "returned". If it's about `old`, then better to name it explicitly, e.g. 
```suggestion
Whether `old` has type `Some{V}(value)` or `Some{typeof(value)}(value)` is an implementation
```